nobl9/release_channel_schema.go

```diff
@@ -0,0 +1,28 @@
+package nobl9
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const releaseChannel = "release_channel"
+
+func schemaReleaseChannel() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Release channel of the created datasource [stable/beta]",
+	}
+}
+
+func marshalReleaseChannel(d *schema.ResourceData) string {
+	return d.Get(releaseChannel).(string)
+}
```

Comment:
I'd parse the channel here and verify it's valid:

```suggestion
func marshalReleaseChannel(d *schema.ResourceData) (string, error) {
	rc, ok := d.Get(releaseChannel).(string)
	if !ok {
		return "", nil
	}
	if _, err := v1alpha.ParseReleaseChannel(rc); err != nil {
		return "", err
	}
	return rc, nil
}
```